internal/common/owner.go

```diff
@@ -26,3 +28,36 @@ func CheckOwnerAnnotation(obj client.Object, owner client.Object) bool {
 	namespacedNameVal, ok := annotations[handler.NamespacedNameAnnotation]
 	return ok && namespacedNameVal == sspNamespacedName.String()
 }
+
+func ListOwnedResources[L any, T any, PtrL interface {
+	*L
+	client.ObjectList
+}, PtrT interface {
+	*T
+	client.Object
+}](request *Request, listOpts ...client.ListOption) ([]T, error) {
```

Comment:
This reads very hard.

Reply:
Better leave it like it is, a single line would make it worse IMO. Also, let's stay away from reflection if we can. :+1:

Reply:
Ok, I will leave it as is for now. 
If you will have any idea how to improve it in the future, please tell me. I don't like that golang requires separate generic parameters for `T` and pointer to `T`.